edn/tests/tests.rs

```diff
@@ -15,207 +15,424 @@ extern crate ordered_float;
 use std::collections::{BTreeSet, BTreeMap, LinkedList};
 use std::iter::FromIterator;
 use std::f64;
+
 use num::bigint::ToBigInt;
 use num::traits::{Zero, One};
 use ordered_float::OrderedFloat;
+
+use edn::parse::{self, ParseError};
+use edn::types::{Value, SpannedValue, Span, ValueAndSpan};
 use edn::symbols;
-use edn::types::Value;
-use edn::types::Value::*;
-use edn::parse::*;
 use edn::utils;
 
-// Helper for making wrapped keywords with a namespace.
-fn k_ns(ns: &str, name: &str) -> Value {
-    NamespacedKeyword(symbols::NamespacedKeyword::new(ns, name))
-}
-
-// Helper for making wrapped keywords without a namespace.
-fn k_plain(name: &str) -> Value {
-    Keyword(symbols::Keyword::new(name))
-}
-
-// Helper for making wrapped symbols with a namespace
-fn s_ns(ns: &str, name: &str) -> Value {
-    NamespacedSymbol(symbols::NamespacedSymbol::new(ns, name))
+// Helpers for parsing strings and converting them into edn::Value.
+macro_rules! fn_parse_into_value {
+    ($name: ident) => {
+        fn $name<'a, T>(src: T) -> Result<Value, ParseError> where T: Into<&'a str> {
+            parse::$name(src.into()).map(|x| x.inner.into())
+        }
+    }
 }
 
-// Helper for making wrapped symbols without a namespace
-fn s_plain(name: &str) -> Value {
-    PlainSymbol(symbols::PlainSymbol::new(name))
-}
+fn_parse_into_value!(nil);
+fn_parse_into_value!(nan);
+fn_parse_into_value!(infinity);
+fn_parse_into_value!(boolean);
+fn_parse_into_value!(bigint);
+fn_parse_into_value!(octalinteger);
+fn_parse_into_value!(hexinteger);
+fn_parse_into_value!(basedinteger);
+fn_parse_into_value!(integer);
+fn_parse_into_value!(float);
+fn_parse_into_value!(text);
+fn_parse_into_value!(symbol);
+fn_parse_into_value!(keyword);
+fn_parse_into_value!(list);
+fn_parse_into_value!(vector);
+fn_parse_into_value!(set);
+fn_parse_into_value!(map);
+fn_parse_into_value!(value);
 
 #[test]
 fn test_nil() {
-    assert_eq!(nil("nil").unwrap(), Nil);
```

Comment:
I think I might be able to get away with a bit less changes, but the crux of the issue is that both SpannedValue and Value are imported the the tests file, and the edn parser returns ValueAndSpan now. Without those changes, we would't be checking Values anymore, but SpannedValues, so some kind of conversion needs to be done: this is via folding, or maintaining the previous intent to test Value, or something else. My approach was one of many possible ones. To make life easier for other people, I chose to drop the `*` import and make things specific.